clueless/models.py

```diff
@@ -114,37 +129,123 @@ class Turn(models.Model):
     """
     player = models.ForeignKey(Player)
     game = models.ForeignKey('Game') #Game class not defined yet, referencing by string
+    status = models.IntegerField(choices=STATUS_CHOICES, default=1)
+
+    def getAvailableActions(self):
+        """
+        :return: Set of Action subclass class objects that can be taken at this point
+        """
+        #TODO: Implement this method
+        return(None)
+
+    def takeAction(self, action):
+        """
+        Performs an action
+        :param action: Subclass of Action, which will have its performAction function called
+        :return:
+        """
+        #TODO: implement this method
+        return(None)
+
+    def endTurn(self):
+        """
+        Ends this turn
+        """
+        pass
 
 class Action(models.Model):
     """
     Any player action a player can take during a turn
     """
-    turn = models.ForeignKey(Turn)
+    turn = models.ForeignKey(Turn, blank=True)
+    description = models.CharField(max_length=255, blank=True)
+    status = models.IntegerField(choices=STATUS_CHOICES, default=1)
+
+    def validate(self):
+        """
+        :return: boolean Can the action be validly taken
+        """
+        raise NotImplementedError("Subclasses of Action need to implement abstract method validate")
```

Comment:
Ah thanks for linking it, yes that does seem the best error looking that list over